src/components/CandidateForm.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from "react";
+
+import styles  from "./CandidateForm.module.css";
+import Button from "./Button";
+
+const CandidateForm = ({ inputs, handleSubmission }) => {
+    //const [data, setData] = useState(inputs.reduce((data, val) => { return {...data, [val.name]: val}}, {}))
+    const [data, setData] = useState(inputs.reduce((data, val) => { return {...data, [val.name] : val.value}}, {}))
```

Comment:
I like the reducer, very slick, but in this case I think we should set the state to the props without trying to format it. 

And then have an onChange() function which gets executed on each keystroke. The onChange function would copy the local state, format and update it accordingly, before setting it to state again.

Typically you would never transform data inside a useState() on component mount.